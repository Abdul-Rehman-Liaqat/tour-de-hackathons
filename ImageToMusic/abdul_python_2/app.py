
from flask import Flask,render_template, url_for, request,redirect
import requests
import numpy as np
import pandas as pd
from subprocess import call

app = Flask(__name__)
#@app.route("/")
#def main():
#    return render_template('send_url.html')

@app.route('/')
def sendData():
    # here we want to get the value of user (i.e. ?url=some-value)
    url_address = request.get_json() 
    print(url_address)
    

if __name__ == '__main__':
      app.run(host='0.0.0.0', port=2222,debug=True)
      