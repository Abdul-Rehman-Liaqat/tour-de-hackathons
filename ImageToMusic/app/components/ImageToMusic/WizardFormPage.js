import React, {Fragment} from 'react'
import {Field, reduxForm} from 'redux-form'
import { connect } from 'react-redux';
import validate from './validate'
import renderField from './renderField'
import moment from 'moment'
import { DatePicker, Form, Input, Radio, Select, Checkbox, Button } from "antd";

const FormItem = Form.Item;
const RadioGroup = Radio.Group;
const { Option } = Select;
const { TextArea } = Input;

const formItemLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 12 }
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 14 }
  }
};

const tailFormItemLayout = {
  wrapperCol: {
    xs: {
      span: 24,
      offset: 0
    },
    sm: {
      span: 14,
      offset: 6
    }
  }
};
const required = value => {
  return (value ? undefined : 'Required')
}
const isNumber = value => {
  return (!isNaN(value) ? undefined : 'Must be number')
}
const makeField = Component => ({ input, meta, children, hasFeedback, label, ...rest }) => {
  const hasError = meta.touched && meta.invalid;
  return (
    <FormItem
      {...formItemLayout}
      label={label}
      validateStatus={hasError ? 'error' : 'success'}
      hasFeedback={hasFeedback && hasError}
      help={hasError && meta.error}
    >
      <Component {...input} {...rest} children={children} />
    </FormItem>
  );
};
const makeDatePickerField = Component => ({ input, meta, children, hasFeedback, label, ...rest }) => {
  const hasError = meta.touched && meta.invalid;
  return (
    <FormItem
      {...formItemLayout}
      label={label}
      validateStatus={hasError ? 'error' : 'success'}
      hasFeedback={hasFeedback && hasError}
      help={hasError && meta.error}
    >
      <Component {...rest} value={input.value !== "" ? moment(input.value, 'YYYY-MM-DD') : input.value} format='YYYY-MM-DD' onChange={(date, dateString) => input.onChange(dateString)} children={children} />
    </FormItem>
  );
};
const AInput = makeField(Input);
const ARadioGroup = makeField(RadioGroup);
const ASelect = makeField(Select);
const ACheckbox = makeField(Checkbox);
const ATextarea = makeField(TextArea);
const ADatePicker = makeDatePickerField(DatePicker);

let WizardFormPage = props => {
  const {handleSubmit,  pristine, reset, submitting, onSubmit, previousPage, nextPage, question, finalSubmit} = props

  return (
      <form onSubmit={handleSubmit}>
        {question[0].type == 'string' &&
          <Field
            name={question[0].id}
            component={AInput}
            label={question[0].text}
            validate={required}
            hasFeedback
          />
        }
        {question[0].type == 'date' &&
          <Field
            name={question[0].id}
            component={ADatePicker}
            label={question[0].text}
            validate={required}
            hasFeedback
          />
        }
        {question[0].type == 'number' &&
          <Field
            name={question[0].id}
            component={AInput}
            label={question[0].text}
            validate={isNumber}
            hasFeedback
          />
        }
        {question[0].type == 'boolean' &&
          <Field
            name={question[0].id}
            component={ARadioGroup}
            label={question[0].text}
            validate={required}
            hasFeedback
          >
            <Radio value="yes">Yes</Radio>
            <Radio value="no">No</Radio>
          </Field>
        }
        <div>
          {previousPage &&
            <Button htmlType="button" className="previous" onClick={previousPage}>
              Previous
            </Button>
          }
          { question[0].next !== null &&
            <Button htmlType="submit" className="next">Next</Button>
          }
          { question[0].next === null &&
            <Button htmlType="submit" onClick={finalSubmit} className="submit">Submit</Button>
          }
        </div>
      </form>
  )
}

WizardFormPage = reduxForm({
  form: 'wizard', // <------ same form name
  destroyOnUnmount: false, // <------ preserve form data
  forceUnregisterOnUnmount: true, // <------ unregister fields on unmount
})(WizardFormPage)

WizardFormPage = connect()(WizardFormPage);
export default WizardFormPage
