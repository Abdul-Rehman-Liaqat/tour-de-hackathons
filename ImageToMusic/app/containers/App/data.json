{
  "questions": [
    {
      "id": "A01",
      "text": "What happened to your product?",
      "reply": "",
      "type": "string",
      "next": "A02"
    },
    {
      "id": "A02",
      "text": "When did the incident occur?",
      "reply": "",
      "type": "date",
      "next": "A04"
    },
    {
      "id": "A03",
      "text": "When was your last claim submission?",
      "reply": "",
      "type": "date",
      "next": "A04"
    },
    {
      "id": "A04",
      "text": "When did you purchase your product?",
      "reply": "",
      "type": "date",
      "next": "A05"
    },
    {
      "id": "A05",
      "text": "How much did your product cost?",
      "reply": "",
      "type": "number",
      "next": "A06"
    },
    {
      "id": "A06",
      "text": "Are you insured elsewhere?",
      "reply": "",
      "type": "boolean",
      "next": "B01"
    },
    {
      "id": "A07",
      "text": "Name of additional insurance?",
      "reply": "",
      "type": "string",
      "next": null
    },
    {
      "id": "B01",
      "text": "Was a police report filed?",
      "reply": "",
      "type": "boolean",
      "next": "C01"
    },
    {
      "id": "C01",
      "text": "Please provide us with your full name?",
      "reply": "",
      "type": "string",
      "next": "C02"
    },
    {
      "id": "C02",
      "text": "Please provide us with a contact number?",
      "reply": "",
      "type": "string",
      "next": "C03"
    },
    {
      "id": "C03",
      "text": "Please provide us with a contact email address?",
      "reply": "",
      "type": "string",
      "next": null
    },
    {
      "id": "C04",
      "text": "Please provide us with your date of birth?",
      "reply": "",
      "type": "date",
      "next": null
    }
  ]
}
