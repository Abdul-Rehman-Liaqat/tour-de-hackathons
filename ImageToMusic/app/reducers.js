/**
 * Combine all reducers in this file and export the combined reducers.
 */

import { combineReducers } from 'redux';
import { routerReducer } from 'react-router-redux';

import globalReducer from 'containers/App/reducer';
import languageProviderReducer from 'containers/LanguageProvider/reducer';
import { reducer as formReducer } from 'redux-form'

/**
 * Creates the main reducer with the dynamically injected ones
 */
export default function createReducer(injectedReducers) {
  return combineReducers({
    routing: routerReducer,
    global: globalReducer,
    language: languageProviderReducer,
    form: formReducer,
    ...injectedReducers,
  });
}
