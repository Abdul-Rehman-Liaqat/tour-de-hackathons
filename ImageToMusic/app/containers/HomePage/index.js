/*
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';
import { FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';
import { Row, Col, Button } from 'antd';
import {change} from 'redux-form'
import injectReducer from 'utils/injectReducer';
import injectSaga from 'utils/injectSaga';
import { makeGlobalState, makeFormValue, makeSelectAnswers, makeSelectQuestions, makeSelectRepos, makeSelectLoading, makeSelectError } from 'containers/App/selectors';
import H2 from 'components/H2';
import ImageToMusic from 'components/ImageToMusic';
import AtPrefix from './AtPrefix';
import CenteredSection from './CenteredSection';
import Form from './Form';
import Input from './Input';
import Section from './Section';
import messages from './messages';
import { finalSubmitForm, loadRepos, saveUserAnswers } from '../App/actions';
import { changeUsername } from './actions';
import { makeSelectUsername } from './selectors';
import reducer from './reducer';
import saga from './saga';

import showResults from './showResults'

export class HomePage extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  /**
   * when initial state username is not null, submit the form to load repos
   */
  componentDidMount() {
    if (this.props.username && this.props.username.trim().length > 0) {
      this.props.onSubmitForm();
    }
  }

  render() {
    const { questions, formValue } = this.props;

    return (
      <div>
        <article>
          <Helmet>
            <title>Home Page</title>
            <meta name="description" content="A React.js Boilerplate application homepage" />
          </Helmet>
        </article>
        <Row style={{ padding: 15, display: 'flex', textAlign: 'center', justifyContent: 'center' }}>
          <Col span={24} >
            <h1>Take Picture and Find songs in your mood!</h1>
          </Col>
        </Row>
        <Row style={{ padding: 15, display: 'flex', textAlign: 'center', justifyContent: 'center' }}>
          <Col span={24} >
            <ImageToMusic />
          </Col>
        </Row>
      </div>
    );
  }
}

HomePage.propTypes = {
  loading: PropTypes.bool,
  error: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.bool,
  ]),
  repos: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.bool,
  ]),
  onSubmitForm: PropTypes.func,
  username: PropTypes.string,
  onChangeUsername: PropTypes.func,
};

export function mapDispatchToProps(dispatch) {
  return {
    dispatch,
    finalSubmitForm: () => dispatch(finalSubmitForm()),
    onChangeUsername: (evt) => dispatch(changeUsername(evt.target.value)),
    onSubmitForm: (evt) => {
      if (evt !== undefined && evt.preventDefault) evt.preventDefault();
      dispatch(loadRepos());
    },
    saveUserAnswers: (answers) => dispatch(saveUserAnswers(answers)),
  };
}

const mapStateToProps = createStructuredSelector({
  globalState: makeGlobalState(),
  formValue: makeFormValue(),
  answers: makeSelectAnswers(),
  questions: makeSelectQuestions(),
  repos: makeSelectRepos(),
  username: makeSelectUsername(),
  loading: makeSelectLoading(),
  error: makeSelectError(),
});
const withConnect = connect(mapStateToProps, mapDispatchToProps);

const withReducer = injectReducer({ key: 'home', reducer });
const withSaga = injectSaga({ key: 'home', saga });

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(HomePage);
