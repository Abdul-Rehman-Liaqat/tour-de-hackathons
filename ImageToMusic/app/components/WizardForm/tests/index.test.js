import { shallow, mount } from 'enzyme';
import React from 'react';
import { Provider } from "react-redux";

jest.mock('react-dom')

import {store} from 'containers/app'
import data from 'containers/App/data.json';
import WizardFormPage from '../WizardFormPage';
import connectedWizardForm, {WizardForm as unConnectedWizardForm} from '../index';

describe('<WizardForm />', () => {
  const props = {
    quesitons: [{
      id: "",
      text: "",
      reply: "",
      type: "",
      next: "",
    }],
    submitted: false,
    finalSubmitForm: jest.fn(),
    formValue: {},
  }
  it('should render one div with .WizardFormPage', () => {
    const renderedComponent = shallow(
      <unConnectedWizardForm />
    );
    expect(renderedComponent.find('.WizardFormPage').length).toBe(1);

  });

  it('should render WizardFormPage firstpage', () => {
    let subject = null
  	let submitting, touched, error, reset, onSave, onSaveResponse, handleSubmit

  	beforeEach(() => {
  		submitting = false
  		touched = false
  		error = null
  		reset = jest.fn()
  		onSaveResponse = Promise.resolve()
  		handleSubmit = fn => fn
  	})
    const props = {
			onSave,
			submitting: submitting,
			// The real redux form has many properties for each field,
			// including onChange and onBlur handlers. We only need to provide
			// the ones that will change the rendered output.
			fields: {
				A01: {
					value: '',
					touched: touched,
					error: error
				}
			},
			handleSubmit,
			reset
		}
    // const firstQuestion = {
    //   id: 'A01',
    //   next: 'A02',
    //   reply: '',
    //   text: 'What happened to your product?',
    //   type: 'string',
    // }
    const html = `<div><ReduxForm page=\"0\" question={{\"id\": \"A01\", \"next\": \"A02\", \"reply\": \"\", \"text\": \"What happened to your product?\", \"type\": \"string\"}} /></div>`
    const text = "What happened to your product?"
    const nextPage = () => {}
    const renderedComponent = shallow(
      <Provider store={store}>
        <connectedWizardForm questions={data.questions} />
      </Provider>
    );
    expect(renderedComponent).toMatchSnapshot();

  });
});
