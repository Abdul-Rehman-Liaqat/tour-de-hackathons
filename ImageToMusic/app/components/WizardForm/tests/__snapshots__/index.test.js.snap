// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WizardForm /> should render WizardFormPage firstpage 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <connectedWizardForm
    fields={
        Object {
            "A01": Object {
              "error": undefined,
              "touched": undefined,
              "value": "",
            },
          }
    }
    handleSubmit={undefined}
    onSave={undefined}
    questions={
        Array [
            Object {
              "id": "A01",
              "next": "A02",
              "reply": "",
              "text": "What happened to your product?",
              "type": "string",
            },
            Object {
              "id": "A02",
              "next": "A04",
              "reply": "",
              "text": "When did the incident occur?",
              "type": "date",
            },
            Object {
              "id": "A03",
              "next": "A04",
              "reply": "",
              "text": "When was your last claim submission?",
              "type": "date",
            },
            Object {
              "id": "A04",
              "next": "A05",
              "reply": "",
              "text": "When did you purchase your product?",
              "type": "date",
            },
            Object {
              "id": "A05",
              "next": "A06",
              "reply": "",
              "text": "How much did your product cost?",
              "type": "number",
            },
            Object {
              "id": "A06",
              "next": "B01",
              "reply": "",
              "text": "Are you insured elsewhere?",
              "type": "boolean",
            },
            Object {
              "id": "A07",
              "next": null,
              "reply": "",
              "text": "Name of additional insurance?",
              "type": "string",
            },
            Object {
              "id": "B01",
              "next": "C01",
              "reply": "",
              "text": "Was a police report filed?",
              "type": "boolean",
            },
            Object {
              "id": "C01",
              "next": "C02",
              "reply": "",
              "text": "Please provide us with your full name?",
              "type": "string",
            },
            Object {
              "id": "C02",
              "next": "C03",
              "reply": "",
              "text": "Please provide us with a contact number?",
              "type": "string",
            },
            Object {
              "id": "C03",
              "next": null,
              "reply": "",
              "text": "Please provide us with a contact email address?",
              "type": "string",
            },
            Object {
              "id": "C04",
              "next": null,
              "reply": "",
              "text": "Please provide us with your date of birth?",
              "type": "date",
            },
          ]
    }
    reset={undefined}
    submitting={undefined}
/>,
  "nodes": Array [
    <connectedWizardForm
      fields={
            Object {
                  "A01": Object {
                    "error": undefined,
                    "touched": undefined,
                    "value": "",
                  },
                }
      }
      handleSubmit={undefined}
      onSave={undefined}
      questions={
            Array [
                  Object {
                    "id": "A01",
                    "next": "A02",
                    "reply": "",
                    "text": "What happened to your product?",
                    "type": "string",
                  },
                  Object {
                    "id": "A02",
                    "next": "A04",
                    "reply": "",
                    "text": "When did the incident occur?",
                    "type": "date",
                  },
                  Object {
                    "id": "A03",
                    "next": "A04",
                    "reply": "",
                    "text": "When was your last claim submission?",
                    "type": "date",
                  },
                  Object {
                    "id": "A04",
                    "next": "A05",
                    "reply": "",
                    "text": "When did you purchase your product?",
                    "type": "date",
                  },
                  Object {
                    "id": "A05",
                    "next": "A06",
                    "reply": "",
                    "text": "How much did your product cost?",
                    "type": "number",
                  },
                  Object {
                    "id": "A06",
                    "next": "B01",
                    "reply": "",
                    "text": "Are you insured elsewhere?",
                    "type": "boolean",
                  },
                  Object {
                    "id": "A07",
                    "next": null,
                    "reply": "",
                    "text": "Name of additional insurance?",
                    "type": "string",
                  },
                  Object {
                    "id": "B01",
                    "next": "C01",
                    "reply": "",
                    "text": "Was a police report filed?",
                    "type": "boolean",
                  },
                  Object {
                    "id": "C01",
                    "next": "C02",
                    "reply": "",
                    "text": "Please provide us with your full name?",
                    "type": "string",
                  },
                  Object {
                    "id": "C02",
                    "next": "C03",
                    "reply": "",
                    "text": "Please provide us with a contact number?",
                    "type": "string",
                  },
                  Object {
                    "id": "C03",
                    "next": null,
                    "reply": "",
                    "text": "Please provide us with a contact email address?",
                    "type": "string",
                  },
                  Object {
                    "id": "C04",
                    "next": null,
                    "reply": "",
                    "text": "Please provide us with your date of birth?",
                    "type": "date",
                  },
                ]
      }
      reset={undefined}
      submitting={undefined}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={undefined}
>
        <connectedWizardForm
                fields={
                        Object {
                                "A01": Object {
                                  "error": undefined,
                                  "touched": undefined,
                                  "value": "",
                                },
                              }
                }
                handleSubmit={undefined}
                onSave={undefined}
                questions={
                        Array [
                                Object {
                                  "id": "A01",
                                  "next": "A02",
                                  "reply": "",
                                  "text": "What happened to your product?",
                                  "type": "string",
                                },
                                Object {
                                  "id": "A02",
                                  "next": "A04",
                                  "reply": "",
                                  "text": "When did the incident occur?",
                                  "type": "date",
                                },
                                Object {
                                  "id": "A03",
                                  "next": "A04",
                                  "reply": "",
                                  "text": "When was your last claim submission?",
                                  "type": "date",
                                },
                                Object {
                                  "id": "A04",
                                  "next": "A05",
                                  "reply": "",
                                  "text": "When did you purchase your product?",
                                  "type": "date",
                                },
                                Object {
                                  "id": "A05",
                                  "next": "A06",
                                  "reply": "",
                                  "text": "How much did your product cost?",
                                  "type": "number",
                                },
                                Object {
                                  "id": "A06",
                                  "next": "B01",
                                  "reply": "",
                                  "text": "Are you insured elsewhere?",
                                  "type": "boolean",
                                },
                                Object {
                                  "id": "A07",
                                  "next": null,
                                  "reply": "",
                                  "text": "Name of additional insurance?",
                                  "type": "string",
                                },
                                Object {
                                  "id": "B01",
                                  "next": "C01",
                                  "reply": "",
                                  "text": "Was a police report filed?",
                                  "type": "boolean",
                                },
                                Object {
                                  "id": "C01",
                                  "next": "C02",
                                  "reply": "",
                                  "text": "Please provide us with your full name?",
                                  "type": "string",
                                },
                                Object {
                                  "id": "C02",
                                  "next": "C03",
                                  "reply": "",
                                  "text": "Please provide us with a contact number?",
                                  "type": "string",
                                },
                                Object {
                                  "id": "C03",
                                  "next": null,
                                  "reply": "",
                                  "text": "Please provide us with a contact email address?",
                                  "type": "string",
                                },
                                Object {
                                  "id": "C04",
                                  "next": null,
                                  "reply": "",
                                  "text": "Please provide us with your date of birth?",
                                  "type": "date",
                                },
                              ]
                }
                reset={undefined}
                submitting={undefined}
        />
</Provider>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <connectedWizardForm
            fields={
                        Object {
                                    "A01": Object {
                                      "error": undefined,
                                      "touched": undefined,
                                      "value": "",
                                    },
                                  }
            }
            handleSubmit={undefined}
            onSave={undefined}
            questions={
                        Array [
                                    Object {
                                      "id": "A01",
                                      "next": "A02",
                                      "reply": "",
                                      "text": "What happened to your product?",
                                      "type": "string",
                                    },
                                    Object {
                                      "id": "A02",
                                      "next": "A04",
                                      "reply": "",
                                      "text": "When did the incident occur?",
                                      "type": "date",
                                    },
                                    Object {
                                      "id": "A03",
                                      "next": "A04",
                                      "reply": "",
                                      "text": "When was your last claim submission?",
                                      "type": "date",
                                    },
                                    Object {
                                      "id": "A04",
                                      "next": "A05",
                                      "reply": "",
                                      "text": "When did you purchase your product?",
                                      "type": "date",
                                    },
                                    Object {
                                      "id": "A05",
                                      "next": "A06",
                                      "reply": "",
                                      "text": "How much did your product cost?",
                                      "type": "number",
                                    },
                                    Object {
                                      "id": "A06",
                                      "next": "B01",
                                      "reply": "",
                                      "text": "Are you insured elsewhere?",
                                      "type": "boolean",
                                    },
                                    Object {
                                      "id": "A07",
                                      "next": null,
                                      "reply": "",
                                      "text": "Name of additional insurance?",
                                      "type": "string",
                                    },
                                    Object {
                                      "id": "B01",
                                      "next": "C01",
                                      "reply": "",
                                      "text": "Was a police report filed?",
                                      "type": "boolean",
                                    },
                                    Object {
                                      "id": "C01",
                                      "next": "C02",
                                      "reply": "",
                                      "text": "Please provide us with your full name?",
                                      "type": "string",
                                    },
                                    Object {
                                      "id": "C02",
                                      "next": "C03",
                                      "reply": "",
                                      "text": "Please provide us with a contact number?",
                                      "type": "string",
                                    },
                                    Object {
                                      "id": "C03",
                                      "next": null,
                                      "reply": "",
                                      "text": "Please provide us with a contact email address?",
                                      "type": "string",
                                    },
                                    Object {
                                      "id": "C04",
                                      "next": null,
                                      "reply": "",
                                      "text": "Please provide us with your date of birth?",
                                      "type": "date",
                                    },
                                  ]
            }
            reset={undefined}
            submitting={undefined}
/>,
          "store": undefined,
        },
        "refs": Object {},
        "state": null,
        "store": undefined,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <connectedWizardForm
          fields={
                    Object {
                              "A01": Object {
                                "error": undefined,
                                "touched": undefined,
                                "value": "",
                              },
                            }
          }
          handleSubmit={undefined}
          onSave={undefined}
          questions={
                    Array [
                              Object {
                                "id": "A01",
                                "next": "A02",
                                "reply": "",
                                "text": "What happened to your product?",
                                "type": "string",
                              },
                              Object {
                                "id": "A02",
                                "next": "A04",
                                "reply": "",
                                "text": "When did the incident occur?",
                                "type": "date",
                              },
                              Object {
                                "id": "A03",
                                "next": "A04",
                                "reply": "",
                                "text": "When was your last claim submission?",
                                "type": "date",
                              },
                              Object {
                                "id": "A04",
                                "next": "A05",
                                "reply": "",
                                "text": "When did you purchase your product?",
                                "type": "date",
                              },
                              Object {
                                "id": "A05",
                                "next": "A06",
                                "reply": "",
                                "text": "How much did your product cost?",
                                "type": "number",
                              },
                              Object {
                                "id": "A06",
                                "next": "B01",
                                "reply": "",
                                "text": "Are you insured elsewhere?",
                                "type": "boolean",
                              },
                              Object {
                                "id": "A07",
                                "next": null,
                                "reply": "",
                                "text": "Name of additional insurance?",
                                "type": "string",
                              },
                              Object {
                                "id": "B01",
                                "next": "C01",
                                "reply": "",
                                "text": "Was a police report filed?",
                                "type": "boolean",
                              },
                              Object {
                                "id": "C01",
                                "next": "C02",
                                "reply": "",
                                "text": "Please provide us with your full name?",
                                "type": "string",
                              },
                              Object {
                                "id": "C02",
                                "next": "C03",
                                "reply": "",
                                "text": "Please provide us with a contact number?",
                                "type": "string",
                              },
                              Object {
                                "id": "C03",
                                "next": null,
                                "reply": "",
                                "text": "Please provide us with a contact email address?",
                                "type": "string",
                              },
                              Object {
                                "id": "C04",
                                "next": null,
                                "reply": "",
                                "text": "Please provide us with your date of birth?",
                                "type": "date",
                              },
                            ]
          }
          reset={undefined}
          submitting={undefined}
/>,
        "_debugID": 2,
        "_renderedOutput": <connectedWizardForm
          fields={
                    Object {
                              "A01": Object {
                                "error": undefined,
                                "touched": undefined,
                                "value": "",
                              },
                            }
          }
          handleSubmit={undefined}
          onSave={undefined}
          questions={
                    Array [
                              Object {
                                "id": "A01",
                                "next": "A02",
                                "reply": "",
                                "text": "What happened to your product?",
                                "type": "string",
                              },
                              Object {
                                "id": "A02",
                                "next": "A04",
                                "reply": "",
                                "text": "When did the incident occur?",
                                "type": "date",
                              },
                              Object {
                                "id": "A03",
                                "next": "A04",
                                "reply": "",
                                "text": "When was your last claim submission?",
                                "type": "date",
                              },
                              Object {
                                "id": "A04",
                                "next": "A05",
                                "reply": "",
                                "text": "When did you purchase your product?",
                                "type": "date",
                              },
                              Object {
                                "id": "A05",
                                "next": "A06",
                                "reply": "",
                                "text": "How much did your product cost?",
                                "type": "number",
                              },
                              Object {
                                "id": "A06",
                                "next": "B01",
                                "reply": "",
                                "text": "Are you insured elsewhere?",
                                "type": "boolean",
                              },
                              Object {
                                "id": "A07",
                                "next": null,
                                "reply": "",
                                "text": "Name of additional insurance?",
                                "type": "string",
                              },
                              Object {
                                "id": "B01",
                                "next": "C01",
                                "reply": "",
                                "text": "Was a police report filed?",
                                "type": "boolean",
                              },
                              Object {
                                "id": "C01",
                                "next": "C02",
                                "reply": "",
                                "text": "Please provide us with your full name?",
                                "type": "string",
                              },
                              Object {
                                "id": "C02",
                                "next": "C03",
                                "reply": "",
                                "text": "Please provide us with a contact number?",
                                "type": "string",
                              },
                              Object {
                                "id": "C03",
                                "next": null,
                                "reply": "",
                                "text": "Please provide us with a contact email address?",
                                "type": "string",
                              },
                              Object {
                                "id": "C04",
                                "next": null,
                                "reply": "",
                                "text": "Please provide us with your date of birth?",
                                "type": "date",
                              },
                            ]
          }
          reset={undefined}
          submitting={undefined}
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={undefined}
>
    <connectedWizardForm
        fields={
            Object {
                "A01": Object {
                  "error": undefined,
                  "touched": undefined,
                  "value": "",
                },
              }
        }
        handleSubmit={undefined}
        onSave={undefined}
        questions={
            Array [
                Object {
                  "id": "A01",
                  "next": "A02",
                  "reply": "",
                  "text": "What happened to your product?",
                  "type": "string",
                },
                Object {
                  "id": "A02",
                  "next": "A04",
                  "reply": "",
                  "text": "When did the incident occur?",
                  "type": "date",
                },
                Object {
                  "id": "A03",
                  "next": "A04",
                  "reply": "",
                  "text": "When was your last claim submission?",
                  "type": "date",
                },
                Object {
                  "id": "A04",
                  "next": "A05",
                  "reply": "",
                  "text": "When did you purchase your product?",
                  "type": "date",
                },
                Object {
                  "id": "A05",
                  "next": "A06",
                  "reply": "",
                  "text": "How much did your product cost?",
                  "type": "number",
                },
                Object {
                  "id": "A06",
                  "next": "B01",
                  "reply": "",
                  "text": "Are you insured elsewhere?",
                  "type": "boolean",
                },
                Object {
                  "id": "A07",
                  "next": null,
                  "reply": "",
                  "text": "Name of additional insurance?",
                  "type": "string",
                },
                Object {
                  "id": "B01",
                  "next": "C01",
                  "reply": "",
                  "text": "Was a police report filed?",
                  "type": "boolean",
                },
                Object {
                  "id": "C01",
                  "next": "C02",
                  "reply": "",
                  "text": "Please provide us with your full name?",
                  "type": "string",
                },
                Object {
                  "id": "C02",
                  "next": "C03",
                  "reply": "",
                  "text": "Please provide us with a contact number?",
                  "type": "string",
                },
                Object {
                  "id": "C03",
                  "next": null,
                  "reply": "",
                  "text": "Please provide us with a contact email address?",
                  "type": "string",
                },
                Object {
                  "id": "C04",
                  "next": null,
                  "reply": "",
                  "text": "Please provide us with your date of birth?",
                  "type": "date",
                },
              ]
        }
        reset={undefined}
        submitting={undefined}
    />
</Provider>,
}
`;
